### PR TITLE
Fix ThemeCollector Symfony 7 compatibility

### PR DESCRIPTION
**Pull request by alexander-schranz**
>  PHP Fatal error:  Type of Sylius\Bundle\ThemeBundle\Collector\ThemeCollector::$data must be Symfony\Component\VarDumper\Cloner\Data|array (as in class Symfony\Component\HttpKernel\DataCollector\DataCollector) in /home/runner/work/SyliusThemeBundle/SyliusThemeBundle/src/Collector/ThemeCollector.php on line 24

I added a very simple test case which just init the ThemeCollector to catch this.

### CODE REVIEW

File: src/Collector/ThemeCollector.php [hunk 1]
```diff
@@ -60,7 +55,7 @@ public function getUsedTheme(): ?ThemeInterface
     }
 
     /**
-     * @return array|ThemeInterface[]
+     * @return ThemeInterface[]
```
Comment (alexander-schranz) [inline src/Collector/ThemeCollector.php:57]: see phpdoc above

File: src/Collector/ThemeCollector.php [hunk 2]
```diff
@@ -29,15 +33,6 @@ final class ThemeCollector extends DataCollector
...
     private ThemeHierarchyProviderInterface $themeHierarchyProvider;
 
-    /**
-     * @var array
-     *
-     * @psalm-var array{used_theme: ?ThemeInterface, used_themes: ThemeInterface[], themes: ThemeInterface[]}
-     *
-     * @psalm-suppress NonInvariantDocblockPropertyType
-     */
-    protected $data;
```
Comment (alexander-schranz) [inline src/Collector/ThemeCollector.php:39]: This variable is defined by DataCollector itself. With beginning of Symfony 7 it has a Union Type here: https://github.com/symfony/symfony/blob/1a16ebc32598faada074e0af12a6a698d2964a5e/src/Symfony/Component/HttpKernel/DataCollector/DataCollector.php#L31

For BC reasons best is to move this variable to PHPDoc definition. So we can support Symfony 7 and previous versions of the parent collector class.

File: composer.json [hunk 3]
```diff
@@ -80,7 +80,9 @@
             "@composer validate --strict",
             "vendor/bin/ecs check spec src tests || true",
             "vendor/bin/psalm",
-            "vendor/bin/rector process --dry-run"
+            "vendor/bin/rector process --dry-run",
+            "tests/Application/bin/console lint:container --env dev",
+            "tests/Application/bin/console lint:container --env prod"
```
Comment (alexander-schranz) [inline composer.json:85]: Stumbled now over that you also have a working console the lint:container would also catch such errors. I recommend always lint for dev and prod container as they may have different set of services.

File: src/Collector/ThemeCollector.php [hunk 4]
```diff
@@ -21,6 +21,10 @@
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\DataCollector\DataCollector;
 
+/**
+ * @property $data array{used_theme: ?ThemeInterface, used_themes: ThemeInterface[], themes: ThemeInterface[]}
+
```
Comment (GSadee) [inline src/Collector/ThemeCollector.php]:
```suggestion
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Alexander! :1st_place_medal:

**Pull request merged**